### DISCUSSION TITLE
set number of GPUs to run monailabel server

### DISCUSSION THREAD

**Discussion by hw-ju** [Project-MONAI/MONAILabel#1400] [Q&A]
Hi!

Can we specify the number of GPUs we want the monailabel server to run on by adding a flag in the `monailabel start_server` command? 

When the server and client, which both require GPUs, run on the same machine, how can we split GPUs to the two?

**Comment by tangy5** [comment]
We can change the use of number GPU and GPU id in the options panel. 
You can enable developer mode, can change the config there.

**Reply by hw-ju** [reply]
Thanks @tangy5 for your super fast response!!

Is that the only way for now? There's no flag that can be passed to `monailabel start_server` to specify the number of GPUs, correct?

If we don't "change the use of number GPU and GPU id in the options panel", the monailabel server will grab all the GPUs available to it, correct?

**Comment by tangy5** [comment]
I think the other way is we follow pytorch's API on specifying this before the start server command:

`export CUDA_VISIBLE_DEVICES=0,1
`
or 

`CUDA_VISIBLE_DEVICES=0,1 <command>
`
You could explore this.

**Reply by hw-ju** [reply]
Thanks a lot @tangy5!
